common/ehub_collector.js

```diff
@@ -31,39 +31,31 @@ const defaultProcessError = function(context, err, messages) {
 module.exports = function (context, eventHubMessages, parseFun, processErrorFun, callback) {
     var processError = processErrorFun ? processErrorFun : defaultProcessError;
     var collector = new AlAzureCollector(context, 'ehub', pkg.version);
-    async.reduce(eventHubMessages, {processed: 0, skipped: 0},
-        function(acc, msgArray, callback) {
+    async.reduce(eventHubMessages, [],
+        function(acc, message, reduceCallback) {
+            return reduceCallback(null,[...acc, ...message.records]);
+        },
+        function(err, redResult) {
             try {
-                collector.processLog(msgArray.records, parseFun, null,
+                collector.processLog(redResult, parseFun, null,
                     function(err) {
                         if (err) {
-                            acc.skipped += processError(context, err, msgArray);
+                            const skipped =  processError(context, err, redResult);
+                            context.log.error(`Error while processing records. Skipped ${skipped} Records`);
+                            return callback(err);
                         } else {
-                            acc.processed += msgArray.records.length;
+                            const processed = redResult.length;
+                            context.log.info(`Processed: ${processed}`);
                         }
-                        return callback(null, acc);
+                        return callback(null, { processed: redResult, skipped: 0 });
```

Comment:
`[]` looks to be the new accumulator. Please double check

Reply:
good spot. its needs to be `redResult.length`